instrumentation/net/http/labeler.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/label"
+)
+
+// Labeler is used to allow instrumented HTTP handlers to add custom labels to
+// the metrics recorded by the net/http instrumentation.
+type Labeler struct {
```

Comment:
Can we add an example test to show how to use this?

Reply:
I added use of the `Labeler` to the existing `Handler` example test since an independent example would look very similar.